### PR TITLE
Tweak Sign Up page

### PR DESCRIPTION
**Pull request by DrShpongle**
Some tweaks to the new Sign Up page:
- different content for the non-logged in and logged in users in header and footer
- tech icons in the header are now linked to related topic pages for logged-in users
- Pricing widget is extracted to the separate component so that we can use it in different places (on Sign Up page in particular)
- adjust Become a Member button in Pricing widget with disabled state and spinner

<details>
  <summary>non-logged in (light):</summary>
  
</details>
<details>
  <summary>non-logged in (dark):</summary>
  
</details>
<details>
  <summary>logged in (light):</summary>
  
</details>
<details>
  <summary>logged in (dark):</summary>
  
</details>

giphy

### CODE REVIEW

File: src/pages/pricing/index.tsx [hunk 1]
```diff
@@ -43,92 +44,93 @@ const Pricing: FunctionComponent<PricingProps> & {getLayout: any} = () => {
     send,
     priceId,
     quantity,
-    prices,
+    // prices,
     availableCoupons,
-    currentPlan,
+    // currentPlan,
   } = useCommerceMachine()
```
Comment (jbranchaud) [inline src/pages/pricing/index.tsx:50]: Since this is completely contained in `PricingWidget` now, it can be deleted from here.

**Comment by joelhooks** [comment]
a logged-in user won't see this because the middleware will show them the curated content page. We need to capture pro status in cookie

I actually think this CTA should be more like the design we used on the video overlay because this looks odd in this format to me, and I think it should be the default at the bottom of the page (no "free account cta" down here if they've read this far)

File: src/components/pages/landing/footer/index.tsx [hunk 2]
```diff
@@ -1,54 +1,205 @@
...
+
+  const formik: FormikProps<FormikValues> = useFormik<FormikValues>({
+    initialValues: {
+      email: viewer?.email ?? '',
+    },
+    validationSchema: Yup.object({
+      email: Yup.string().email('Invalid email').required('Required'),
+    }),
+    onSubmit: async () => {
+      track('clicked join cta on signup page')
```
Comment (joelhooks) [inline src/components/pages/landing/footer/index.tsx:35]:
```suggestion
      track('checkout: submit email on signup page')
```

File: src/components/pages/landing/footer/index.tsx [hunk 3]
```diff
@@ -1,54 +1,205 @@
...
+
+      if (hasProAccess) {
+        const message = `You've already got a pro account at ${formik.values.email}. Please sign in.`
+
+        toast.error(message, {
+          duration: 6000,
+          icon: '❌',
+        })
+
+        track('cta overlay checkout: existing pro account found', {
```
Comment (joelhooks) [inline src/components/pages/landing/footer/index.tsx:97]:
```suggestion
        track('checkout: existing pro account found', {
```

File: src/components/pages/landing/footer/index.tsx [hunk 4]
```diff
@@ -1,54 +1,205 @@
...
+          email: formik.values.email,
```
Comment (joelhooks) [inline src/components/pages/landing/footer/index.tsx:98]:
```suggestion
          email: formik.values.email,
          location: 'signup page'
```

File: src/components/pages/landing/footer/index.tsx [hunk 5]
```diff
@@ -1,54 +1,205 @@
...
+
+        // email is already associated with a pro account, return early instead
+        // of sending the user to a Stripe Checkout Session.
+        return
+      }
+    }
+
+    if (emailIsValid(formik.values.email)) {
+      await track('checkout: valid email present', {
+        priceId: priceId,
```
Comment (joelhooks) [inline src/components/pages/landing/footer/index.tsx:109]:
```suggestion
        priceId: priceId,
        location: 'signup page'
```

File: src/components/pages/landing/footer/index.tsx [hunk 6]
```diff
@@ -1,54 +1,205 @@
...
+      })
+      await track('checkout: redirect to stripe', {
+        priceId,
```
Comment (joelhooks) [inline src/components/pages/landing/footer/index.tsx:112]:
```suggestion
        priceId,
        location: 'signup page'
```

File: src/components/pages/landing/footer/index.tsx [hunk 7]
```diff
@@ -1,54 +1,205 @@
...
+        email: formik.values.email,
+        stripeCustomerId: account?.stripe_customer_id,
+        authToken,
+        quantity,
+        coupon: state.context.couponToApply?.couponCode,
+      })
+    } else {
+      // we don't have a valid email for the checkout
+      await track('checkout: unable to proceed, no valid email', {
+        email: formik.values.email,
```
Comment (joelhooks) [inline src/components/pages/landing/footer/index.tsx:126]:
```suggestion
        email: formik.values.email,
        location: 'signup page'
```

**Review by joelhooks — APPROVED** [review]
made some minor analytics changes

**Pull request merged**